Update Basic MQTT Version

djnewbould opened the pull request:
Previously, the Basic MQTT client and GSS MQTT client used different MQTT protocol versions, leading to inconsistency. This update ensures both clients now use the same default MQTT protocol version for uniform behaviour.

djnewbould committed:
Update basic MQTT version

Previously, the Basic MQTT client and GSS MQTT client used different MQTT protocol versions, leading to inconsistency. This update ensures both clients now use the same default MQTT protocol version for uniform behaviour.

amrc-benmorrow reviewed (changes requested):
Can this be the other way around, i.e. `protocolVersion: 5, ...opts`, as this leaves the option to request MQTT3 if needed. (This change could probably be made to the GSSAPI half too; GSSAPI won't work with MQTT3 but might with MQTT6 or whatever's next...)

djnewbould committed:
Update MQTT Opts Ordering

Moved MQTT opts below protocolVersion to allow for other MQTT protocol versions to be requested.
